### DISCUSSION TITLE
Create - User moderation

### DISCUSSION THREAD

**Discussion by davehakkens** [ONEARMY/community-platform#3235] [Big Updates]
With moderation we mean the accepting of new content, the request to the content creator to make changes or the removal of unwanted content. This all to make sure it fits community guidelines

The goal here is to identify the problems and discuss/share possible ways forward to see if it's worth putting on our roadmap.

# Current problems
1. All moderation is done by the team which is not very scalable as the community grows
2. With a growing community its getting hard to see everything
3. Not easy (cheap) to invite external moderators to (re) tool

# Things to keep in mind
- There are many things to moderate (profiles, map-pin, comments, how-tos content, research, questions etc) try to understand what is most relevant
- This is not a big problem yet, but anticipating as the community grows
- Besides the tool its also important that each instance has clear community rules

Whats needed next
- [ ] Identify what kind of moderation flow is needed
- [ ] Design a moderation method
- [ ] Review technical complexity

**Comment by davehakkens** [comment]
First sketch of possible moderation flow
Initial thought: Seems like a big piece of infrastructure to intergrate. Anyone else sees simpler ways?
( we also have the figma file in case you want it)


![1  Flagging](https://github.com/ONEARMY/community-platform/assets/13672737/258e31c9-09a8-479a-9381-1a2efee4f49f)
![2  Report](https://github.com/ONEARMY/community-platform/assets/13672737/7947a09b-6f09-48b8-81f8-b8feeb60ef92)
![3  moderator](https://github.com/ONEARMY/community-platform/assets/13672737/18634dc9-241c-44d1-ac25-5530925dc397)
![4  flow](https://github.com/ONEARMY/community-platform/assets/13672737/4dd27d59-5468-4cf1-b747-6be7f0ce1b00)

**Reply by pizzaisdavid** [reply]
I think in your flow and user interface should be adjusted so a moderator can "assign" themselves to a report (and filter for tickets they assigned to.) Because, otherwise eventually, it will be hard to remember which ones they need to follow-up on.

**Reply by pizzaisdavid** [reply]
> Initial thought: Seems like a big piece of infrastructure to intergrate. Anyone else sees simpler ways?

One way, at least for prototyping or a "first version," would be to use a shared email inbox. This could lead to some mistakes, especially if people don't follow a defined process together, but the whole moderation panel would be skipped for now and the programming work would be lighter.

The flow:
A user wants to report content. They click the report button, it opens the contact form (shown in screenshot 2.) When the user presses submit, it sends two email messages to the shared email address.

The first email message is used as the internal notes, it contain: the reporters information, reason, message, and link. The 'reply-to' field is setup either as the reporter's email address or as the shared email address, ensuring the reporters information is not accidentally sent to someone else. The title would be something like `Internal notes: <offenders_username> <report_reason> (ID: <some_random_id>)`

The second email message is setup to respond to the offender, it contains: the link and some template, which the admin can adjust to their needs. This could depend on the report reason. The 'reply-to' field is the offender's email address. The title would be something like: `User report: <report_reason> (ID: <some_random_id>)`

Both email titles include <some_random_id>, which is the same between them, which makes it possible to find the corresponding internal notes for a user report.

Example email inbox:
```
...
Internal notes: @Jim, dead link (ID: abc123)
User report: dead link (ID: abc123)                                                     [labels: Pending]
Internal notes: @Sarah, offensive content (ID: def456)
User report: offensive content (ID: def456)                                             [labels: In-Review, Tim]
Internal notes: @Bob, other (ID: xyz789)
User report: other (ID: xyz789)                                                         [labels: Resolved, Eva]
...
```

Let's assume you use Gmail as an email provider.

You would setup labels for Pending, In-Review, and Resolved. Email messages can have multiple labels applied, so I'd also suggest also creating one label for each moderator's name, and then when they take ownership over a report, they also assign their label (and also remove the Pending label and add either In-Review or Resolved label.)

Then they can search for emails with their label and In-Review.

I believe in Gmail, you can setup some automated rules, like "when a new email message comes in, if it contains the text 'User report' then automatically add the label Pending.

Downsides:
* It is weak against misuse (someone could just delete an email message, either accidentally or on purpose) -- so as a prevention against this would be: four emails could actually be sent (duplicates sent to a back-up email, and occasionally a someone trusted could compare the total counts.)
* A separate email address feels "far away" moderators probably will forget to check this.
* Needs initially setup for each deployment.
* Logging into an email address from many different locations from many different machines may trigger Google's security checks, and people would be asked to 2-factor authenticate on a device they don't necessarily have access to. (Maybe a different email provider should be picked?)

Upsides:
* If a moderator quits, someone else can easily take over their tickets, because the communication is happening in a central place.
* Email templates will make the work easier.

Because your flow (image 4) doesn't take any automated actions on the platform (sending notifications, disabling content) this proposal can do everything you want (at least as I understood it.)

I would view this as a possible middle step, being replaced by something better in the future.

**Reply by pizzaisdavid** [reply]
To make things more complicated, I also have the question: what should happen to the content while it is In-Review?

Like a broken link on a profile could result in a banner at the top that says something like: "information may be out-of-date"

A how-to with bad language could be reverted to draft (and only publishable upon admin approval.)

Map-pins that have missing information, broken links, or no longer active be grayed out (or something) which kind of tackles this concern of keeping the map up-to-date: https://discord.com/channels/586676777334865928/1219310171470172291/1219310171470172291
